extensions/data-plane/data-plane-azure-storage/src/main/java/org/eclipse/edc/connector/dataplane/azure/storage/pipeline/AzureStorageDataSource.java

```diff
@@ -57,16 +57,17 @@ public void close() {
 
     private AzureStoragePart getPart() {
         try {
-            var adapter = blobStoreApi.getBlobAdapter(accountName, containerName, blobName, sharedKey);
-            return new AzureStoragePart(adapter);
+            return sharedKey != null ?
+                new AzureStoragePart(blobStoreApi.getBlobAdapter(accountName, containerName, blobName, sharedKey))
```

Comment:
Cant we use a Builder pattern for AzureStoragePart? To avoid this if condition

Reply:
@bscholtes1A - Yes, I totally agree with you about that BlobStoreApi, I struggled with that one, too. I just got a message that the code coverage is down by 1% (new code). I guess one reason might be that I changed everything to "var", even if it does not touch the actual functionality of this PR, and the BlobStoreApi is covered only 1/3. I can offer to write a unit test for that part, but if you want to refactor it anyhow, that would not make sense. Just let me know if you want me to do so anyhow. Regarding your proposal: I applied your suggestions but did not commit it yet, just waiting for other reviewers.